### ISSUE TITLE
Version string replacement

### ISSUE THREAD

**Issue by patrickjamesbarry** [dotnet/cli#3365]
Right now, dotnet supports the ability to change pre-release part of sem ver (1.0.0-*).  So the \* is replaced with the value you specify with --version-suffix flag.  This allows you to create packages that nuget will consider Pre-release packages.  We work in a continuous delivery model where each build could be a possibly release candidate if it passes all tests.  Because of that, and after reading the SemVer spec, it seems that our version should be 1.0.0+<build#>.  But dotnet will not accept   

> "version": "1.0.0+*".  

What would be really nice is a looser string replacement of the wildcard *. So instead of only being able to use \* in the prerelease slot, we could put it also in the PATCH column.  

> ie 1.0.*

So basically what I am looking for is the ability to produce a nuget package that is not considered prerelease, and allows me to set part of the version identifier based on the string I pass in with --version-suffix flag.

**Comment by piotrpMSFT** [comment]
being able to better manipulate the version string globally is an interesting feature which we should eventually get to.

**Comment by livarcocc** [comment]
You can now just set a <VersionPrefix> property in msbuild, like this `<VersionPrefix>1.0.0</VersionPrefix>` and then in your CI, if you set VersionSuffix, the right pre-release version will be generated. Otherwise, you will get a release version 1.0.0.
